### PR TITLE
Label update automation

### PR DESCRIPTION
**Pull request by Soullivaneuh**
The goal of this PR is to update all labels of all sonata project.

For the moment, it just output what it should do.

@sonata-project/contributors Please take profits to discuss which label should be added on the list and why (just for label, not color).

This should be done now because all other not wanted labels **will be destroyed**.

You can see the output on Travis jobs.

- [x] Make update real (test only on `cache` and `exporter` for now)
- [x] Only show updated/deleted label
- [x] Success message at end

### CODE REVIEW

**Comment by Th3Mouk** [comment]
@Soullivaneuh what do think about `help needed`. I remember when some tests not passed and i was blocked during lot of week, the community can't distinguish.

Can you also explained `vendor` and `rtm` please ?

I don't know if it's the same think of `documentation`, for PR on the doc.

**Comment by Soullivaneuh** [comment]
> what do think about `help needed`. I remember when some tests not passed and i was blocked during lot of week, the community can't distinguish.

:+1: 

> Can you also explained `vendor` and `rtm` please ?

`vendor`: When the issue/PR is related to the vendor. Example: One issue appear on SonataUserBundle but it's caused by a bug introduced on a new version of FOSUserBundle.

RTM: Ready to Merge. This should be used for example when a PR is OK but CI is not finished. In this case, put this label so you can remember to merge later if CI passes.

> I don't know if it's the same think of documentation, for PR on the doc.

:+1: for that but I think I would name it `doc` only because it's already used by many projects of sonata.

**Comment by core23** [comment]
- docs
- review required - someone else should verify the PR
- test missing (?)

**Comment by greg0ire** [comment]
I'd like a label for cross-validation : I'm reluctant to merge things myself if I'm the only reviewer. `review_required` does not really mean that, does it. I would like a label to mean "ok for me, if it's ok for someone else then merge".

**Comment by Soullivaneuh** [comment]
> test missing (?)

Would not be enough to tell user to add test and add "Pending author" label?

File: src/Config/Configuration.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ */
+class Configuration implements ConfigurationInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getConfigTreeBuilder()
+    {
+        $treeBuilder = new TreeBuilder();
+        $rootNode = $treeBuilder->root('sonata');
```
Comment (OskarStark) [inline src/Config/Configuration.php]: should be `sonata_dev_kit` IMO

**Comment by OskarStark** [comment]
what about `WIP` ?

File: src/Console/Command/DispatchCommand.php [hunk 2]
```diff
@@ -45,13 +64,94 @@ protected function configure()
...
+
+        $configs = Yaml::parse(file_get_contents(__DIR__.'/../../../.sonata-project.yml'));
+        $processor = new Processor();
+        $this->configs = $processor->processConfiguration(new Configuration(), array('sonata' => $configs));
+
+        $this->packagistClient = new \Packagist\Api\Client();
+
+        $this->githubClient = new \Github\Client(
+            new \Github\HttpClient\CachedHttpClient(array(
+                'cache_dir' => sys_get_temp_dir().'/github-api-cache',
```
Comment (greg0ire) [inline src/Console/Command/DispatchCommand.php]: A good practice I found by suffering from crowded `/tmp/` directories on my servers is to also include the name of the project that generates stuff and sometimes does not clean afterwards. So using `/sonata-dev-kit/github-api-cache` would be better IMO

File: src/Console/Command/DispatchCommand.php [hunk 3]
```diff
@@ -45,13 +64,94 @@ protected function configure()
...
+        $missingLabels = $configuredLabels;
+
+        $headers = array('Name', 'Actual color', 'Needed Color', 'State');
+        $rows = array();
+
+        foreach ($this->githubClient->repository()->labels()->all('sonata-project', $repositoryName) as $label) {
+            $name = $label['name'];
+            $color = $label['color'];
+
+            $shouldExists = array_key_exists($name, $configuredLabels);
```
Comment (greg0ire) [inline src/Console/Command/DispatchCommand.php]: `shouldExist`

File: src/Console/Command/DispatchCommand.php (hunk 2 above)
Comment (Soullivaneuh) [inline src/Console/Command/DispatchCommand.php]: Not relevant anymore. Cache causes trouble on execution, removed it for now.

File: src/Config/Configuration.php (hunk 1 above)
Comment (Soullivaneuh) [inline src/Config/Configuration.php]: Not a big deal, this is not used on the yaml file.

**Comment by Soullivaneuh** [comment]
> what about `WIP` ?

@OskarStark You mean this one? https://github.com/sonata-project/SonataAdminBundle/labels/in%20progress

**Comment by Soullivaneuh** [comment]
I am not sure for the "In progress" one because the contributor can't set it himself.

Adding a `[WIP]` prefix on title would be simpler IMHO.

**Comment by OskarStark** [comment]
> @OskarStark You mean this one? https://github.com/sonata-project/SonataAdminBundle/labels/in%20progress

yes, `in progress` is fine, too

**Comment by OskarStark** [comment]
What do you think of "mapping" existing labels instead of only deleting them?
Could it be useful?

**Comment by Soullivaneuh** [comment]
> What do you think of "mapping" existing labels instead of only deleting them?

This is an open door of lot of duplicate and/or unused labels.

If we do that, so would be better to do nothing.

With this we will have consistency. If a label should be added, we can open PR later to discuss it.

**Comment by Th3Mouk** [comment]
no tags when help is needed ?

**Comment by Soullivaneuh** [comment]
Oh sorry @Th3Mouk I forgot it!

**Pull request merged**